### PR TITLE
[MM-40133] Add support for default notifications for followers

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
Add support for default notifications for followers:
* Run start
Screen Shot 2021-11-16 at 17 27 39

* Run finish
Screen Shot 2021-11-16 at 18 02 15

* Status update
Screen Shot 2021-11-16 at 18 04 59

* Status update missed
* Retrospective published
Screen Shot 2021-11-16 at 18 03 12

#### Ticket Link
  Fixes: https://mattermost.atlassian.net/browse/MM-40133

#### Checklist

- ~~[ ] Telemetry updated~~
- ~~[ ] Gated by experimental feature flag~~
- ~~[ ] Unit tests updated~~

### CODE REVIEW

File: server/app/reminder.go [hunk 1]
```diff
@@ -108,12 +108,43 @@ func (s *PlaybookRunServiceImpl) handleStatusUpdateReminder(playbookRunID string
...
+	if s.pluginAPI.Configuration.GetConfig().ServiceSettings.SiteURL != nil {
+		siteURL = *s.pluginAPI.Configuration.GetConfig().ServiceSettings.SiteURL
+	}
+
+	team, err := s.pluginAPI.Team.Get(channel.TeamId)
+	if err != nil {
+		return "", errors.Wrapf(err, "can't get team - %s", channel.TeamId)
+	}
+
+	message := fmt.Sprintf("Status update is overdue for run [%s](%s/%s/channels/%s?telem=todo_overduestatus_clicked&forceRHSOpen) (Owner: @%s)\n",
```
Comment (iomodo) [inline server/app/reminder.go:142]: Notice that clicking on the overdue status update will be counted in the telemetry. Not sure if it's the correct behaviour @stephenvanhemmen

Comment (stephenvanhemmen) [reply server/app/reminder.go:142]: What telemetry event is this measuring? Is it the event for specifically clicking the link in the overdue status update message?

Comment (iomodo) [reply server/app/reminder.go:142]: That's the one

**Comment by iomodo** [comment]
Updated messages as per conversation [here](https://community-daily.mattermost.com/core/pl/k5ibhrzm3inadfonxpd73i6m4e)

**Review by cpoile — APPROVED** [review]
Awesome, thanks @iomodo!

**Review by stephenvanhemmen — APPROVED** [review]
Thanks!

File: server/app/playbook_run_service.go [hunk 2]
```diff
@@ -363,15 +367,24 @@ func (s *PlaybookRunServiceImpl) CreatePlaybookRun(playbookRun *PlaybookRun, pb
...
 
 	if pb != nil {
-		for _, broadcastChannelID := range pb.BroadcastChannelIDs {
-			err = s.broadcastPlaybookRunCreation(pb.Title, pb.ID, broadcastChannelID, playbookRun, owner)
-			if err != nil {
-				s.pluginAPI.Log.Warn("failed to broadcast the playbook run creation to channel", "ChannelID", playbookRun.BroadcastChannelIDs, "error", err)
+		message, err := s.buildPlaybookRunCreationMessage(pb.Title, pb.ID, playbookRun, owner)
+		if err != nil {
+			s.pluginAPI.Log.Warn("failed to build the playbook run creation message", "error", err)
+		} else {
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply server/app/playbook_run_service.go:373]: +1

File: server/app/playbook_run_service.go [hunk 3]
```diff
@@ -863,6 +889,26 @@ func (s *PlaybookRunServiceImpl) OpenFinishPlaybookRunDialog(playbookRunID, trig
...
+	)
+	announcementMsg += fmt.Sprintf(
+		"@%s just marked [%s](%s) as finished. Visit the link above for more information.",
+		userName,
+		playbookRun.Name,
+		getRunDetailsURL(*siteURL, s.configService.GetManifest().Id, playbookRun.ID),
+	)
+
+	return announcementMsg, nil
+}
```
Comment (agarciamontoro) [inline server/app/playbook_run_service.go:910]: We can use the new functions to build relative URLs (added in https://github.com/mattermost/mattermost-plugin-playbooks/pull/875) here and we can get rid of `siteURL`.

File: server/app/reminder.go [hunk 4]
```diff
@@ -108,12 +108,43 @@ func (s *PlaybookRunServiceImpl) handleStatusUpdateReminder(playbookRunID string
...
+	if s.pluginAPI.Configuration.GetConfig().ServiceSettings.SiteURL != nil {
+		siteURL = *s.pluginAPI.Configuration.GetConfig().ServiceSettings.SiteURL
+	}
+
+	team, err := s.pluginAPI.Team.Get(channel.TeamId)
+	if err != nil {
+		return "", errors.Wrapf(err, "can't get team - %s", channel.TeamId)
+	}
+
+	message := fmt.Sprintf("Status update is overdue for [%s](%s/%s/channels/%s?telem=todo_overduestatus_clicked&forceRHSOpen) (Owner: @%s)\n",
```
Comment (agarciamontoro) [inline server/app/reminder.go:142]: We can probably use a relative URL here and also get rid of the `siteURL`.

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Looks great! I just added a couple of comments to use relative URLs instead of absolute ones.

**Review by agarciamontoro — APPROVED** [review]
Looks great, thank you!

**Pull request merged**